### PR TITLE
Provide Spring Boot Starter: xsuaa-spring-starter

### PR DESCRIPTION
**Pull request by nenaraab**
With this PR application developers can include spring-xsuaa library with all dependencies with one dependency and leverage auto-configuation as documented [here](https://github.com/SAP/cloud-security-xsuaa-integration/tree/spring-starter/spring-xsuaa#maven-dependencies).

- These samples are adapted accordingly:
  - spring-security-basic-auth
  - spring-security-xsuaa-usage

Furthermore this PR adds Spring-Context tests to the two above sample apps.

Auto-configuration is only enabled, when applications make use of Spring Boot starters. This change was necessary in order to support plain Spring projects (see also [here](https://github.com/SAP/cloud-security-xsuaa-integration/issues/112)). The incompatibility is documented as well.

### CODE REVIEW

**Comment by nenaraab** [comment]
@mwdb and another one for review :-)

**Comment by firnkes** [comment]
looks good to me.

**Comment by nenaraab** [comment]
Hi @TheFonz2017,
please my adaptions :-)
Best regards,
Nena

File: spring-xsuaa/pom.xml [hunk 1]
```diff
@@ -42,7 +42,7 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-autoconfigure</artifactId>
 			<version>${spring.boot.version}</version>
-			<scope>provided</scope>
+			<optional>true</optional>
```
Comment (mwdb) [inline spring-xsuaa/pom.xml]: Hm...this adds a dependency to a Spring version.  This implies every time Spring updates a new version, this lib also needs to update. Why is this needed?
